src/ng/directive/ngOptions.js

```diff
@@ -428,6 +428,9 @@ var ngOptionsDirective = ['$compile', '$document', '$parse', function($compile,
         }
       }
 
+      // Remove all the elements inside the select.
+      selectElement.empty();
```

Comment:
Please add something like:
"The empty option will be compiled and rendered before we first generate the options"